### PR TITLE
[Caffe Frontend] Add support for Power layer

### PR DESCRIPTION
**Pull request by KennyTang1988**
Add support for Power layer

### CODE REVIEW

File: python/tvm/relay/frontend/caffe.py [hunk 1]
```diff
@@ -637,6 +638,22 @@ def convert_embed(self, op):
...
+        inputs = op.bottom
+        in_expr = self.exp_tab.get_expr(inputs[0])
+        power = _expr.const(op.power_param.power)
+        scale = _expr.const(op.power_param.scale)
+        shift = _expr.const(op.power_param.shift)
+
+        if scale != 1.0:
+            out = _op.multiply(in_expr, scale)
+        if shift != 0.0:
+            out = _op.add(out, shift)
```
Comment (mshr-h) [inline python/tvm/relay/frontend/caffe.py:652]: if `scale` is 1.0, line 650 won't be executed.
Then `out` is undefined which causes an undefined reference error.

File: python/tvm/relay/frontend/caffe.py [hunk 2]
```diff
@@ -637,6 +638,22 @@ def convert_embed(self, op):
...
+        if power != 1.0:
+            out = _op.power(out, power)
```
Comment (mshr-h) [inline python/tvm/relay/frontend/caffe.py:654]: Same here.

File: python/tvm/relay/frontend/caffe.py [hunk 3]
```diff
@@ -637,6 +638,22 @@ def convert_embed(self, op):
...
+        return out
```
Comment (mshr-h) [inline python/tvm/relay/frontend/caffe.py:655]: If `scale` is 1.0 and `shift` is 0.0 and `power` is 1.0, `out` will be undefined.

Comment (mshr-h) [reply python/tvm/relay/frontend/caffe.py:655]: I think we don't need to check the parameters.
Just like below.

```python
out = _op.multiply(in_expr, scale)
out = _op.add(out, shift)
out = _op.power(out, power)
return out
```

File: tests/python/frontend/caffe/test_forward.py [hunk 4]
```diff
@@ -613,6 +613,24 @@ def test_forward_Pooling():
     _test_pooling(data, pool=P.Pooling.AVE, global_pooling=True)
 
 
+#######################################################################
+# Power
+# -----------
```
Comment (mshr-h) [inline tests/python/frontend/caffe/test_forward.py:618]: Can you remove unnecessary redundant `--` and keep aligned with Power?

```python
#######################################################################
# Power
# -----
```

File: tests/python/frontend/caffe/test_forward.py [hunk 5]
```diff
@@ -613,6 +613,24 @@ def test_forward_Pooling():
...
+
+def test_forward_Power():
+    """Power"""
+    data = np.random.rand(1, 3, 10, 10).astype(np.float32)
+    _test_power(data, power_param={"power": 0.37, "scale": 0.83, "shift": -2.4})
+    _test_power(data, power_param={"power": 0.37, "scale": 0.83, "shift": 0.0})
+    _test_power(data, power_param={"power": 0.0, "scale": 0.83, "shift": -2.4})
+    _test_power(data, power_param={"power": 1.0, "scale": 0.83, "shift": -2.4})
+    _test_power(data, power_param={"power": 2.0, "scale": 0.34, "shift": -2.4})
+
```
Comment (mshr-h) [inline tests/python/frontend/caffe/test_forward.py:632]: Might be better to add a test with Caffe's default parameters.

```python
_test_power(data, power_param={"power": 1.0, "scale": 1.0, "shift": 0.0})
```

**Comment by mshr-h** [comment]
@fernchen @FrozenGene 
Can you take a look at this PR, please? Thanks!

**Comment by KennyTang1988** [comment]
> @fernchen @FrozenGene Can you take a look at this PR, please? Thanks!

I hava update commit, please help check it again.

**Review by masahi — APPROVED** [review]

**Pull request merged**